### PR TITLE
Fix for TLS v1.1 length sanity check for large messages

### PR DESCRIPTION
**Pull request by dgarske**
# Description

* Fix for TLS v1.1 length sanity check for large messages (messages within 16 bytes of max TLS payload).
* Fixes for TLS benchmarking app (tls_bench) for shutdown and formatting brace cleanups.
Reported by @anhu 

# Testing

```
./configure && make
./examples/benchmark/tls_bench
```

Was getting error with TLS v1.1 ciphers with large packets due to block and explicit IV size calculation.

```
Verify MAC of Encrypted Data
Plaintext too long - Encrypt-Then-MAC
wolfSSL error occurred, error = -328
wolfSSL Leaving wolfSSL_read_internal(), return -328
error on server read
```

# Checklist

 - [ ] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

File: examples/benchmark/tls_bench.c [hunk 1]
```diff
@@ -523,6 +528,9 @@ static int ClientMemRecv(info_t* info, char* buf, int sz)
 
     pthread_mutex_unlock(&info->to_client.mutex);
 
+    if (info->to_server.done != 0)
```
Comment (anhu) [inline examples/benchmark/tls_bench.c:531]: might as well add braces here?

File: examples/benchmark/tls_bench.c [hunk 2]
```diff
@@ -523,6 +528,9 @@ static int ClientMemRecv(info_t* info, char* buf, int sz)
...
+        return -1;
+
 #ifdef BENCH_USE_NONBLOCK
     if (sz == 0)
```
Comment (anhu) [inline examples/benchmark/tls_bench.c:535]: Brace here too?

File: src/internal.c [hunk 3]
```diff
@@ -17793,13 +17793,15 @@ int ProcessReplyEx(WOLFSSL* ssl, int allowSocketErr)
...
+                >= ssl->buffers.inputBuffer.length) {
                 return BUFFER_ERROR;
-
+            }
        #if defined(HAVE_ENCRYPT_THEN_MAC) && !defined(WOLFSSL_AEAD_ONLY)
             if (IsEncryptionOn(ssl, 0) && ssl->options.startedETMRead) {
-                if ((ssl->curSize -
-                        ssl->keys.padSz -
+                /* For TLS v1.1 the block size and explcit IV are added to idx,
+                 * so it needs included in this limit check */
```
Comment (anhu) [inline src/internal.c:17802]: needs included --> needs to be included

File: src/internal.c [hunk 4]
```diff
@@ -17816,8 +17818,11 @@ int ProcessReplyEx(WOLFSSL* ssl, int allowSocketErr)
             else
        #endif
                 /* TLS13 plaintext limit is checked earlier before decryption */
+                /* For TLS v1.1 the block size and explcit IV are added to idx,
+                 * so it needs included in this limit check */
```
Comment (anhu) [inline src/internal.c:17822]: needs included --> needs to be included

**Review by anhu — COMMENTED** [review]
Thanks for this fix @dgarske !! Great work! 

I still need to try it out.

**Comment by anhu** [comment]
`tls_bench` works!! 
```
./configure --enable-psk --enable-sp CFLAGS=-DWOLFSSL_STATIC_PSK
make all check
examples/benchmark/.libs/tls_bench
```
....yields no errors when testing this change against https://github.com/anhu/wolfssl/tree/bench_psk

**Review by anhu — APPROVED** [review]
Everything works as expected now.

**Review by SparkiDev — APPROVED** [review]

**Pull request merged**